File: core/api/src/test/java/ru/endlesscode/rpginventory/configuration/ConfigurationTest.java
package ru.endlesscode.rpginventory.configuration;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import ru.endlesscode.rpginventory.FileTestBase;
import ru.endlesscode.rpginventory.misc.TestConfiguration;

import java.nio.file.Files;

public class ConfigurationTest extends FileTestBase {

    private ConfigurationProvider<TestConfiguration> testConfiguration;

    @Before
    @Override
    public void setUp() throws Exception {
        super.setUp();
        this.testConfiguration = new ConfigurationProvider<>(this.tmpDir, TestConfiguration.class);
    }

    @Test()
    public void hasConfigurationSaved() {
        assert Files.exists(this.tmpDir.resolve("testConfiguration.conf"));
    }

    @Test
    public void isConfigObjectLoadedProperly() {
        final TestConfiguration config = this.testConfiguration.getConfig();
        Assert.assertNotNull(config);
    }

    @Test
    public void isConfigVariablesLoadedProperly() {
        Assert.assertNotNull(this.testConfiguration);

        final TestConfiguration local = new TestConfiguration();
        final TestConfiguration config = this.testConfiguration.getConfig();
        Assert.assertEquals(local.getaString(), config.getaString());
        Assert.assertEquals(local.getAnInt(), config.getAnInt());
    }

    @Test
    public void variablesEditTest() {
        final String newString = "Lorem ipsum dolor sit amet, consectetur.";

        TestConfiguration config = this.testConfiguration.getConfig();
        config.setAnInt(6);
        config.setaString(newString);
        this.testConfiguration.save();
        this.testConfiguration.reload();

        config = this.testConfiguration.getConfig();
        Assert.assertEquals(newString, config.getaString());
        Assert.assertEquals(6, config.getAnInt());
    }
}

File: core/api/src/test/java/ru/endlesscode/rpginventory/configuration/ConfigurationTest.kt
package ru.endlesscode.rpginventory.configuration

import ru.endlesscode.rpginventory.FileTestBase
import ru.endlesscode.rpginventory.misc.TestConfiguration
import java.nio.file.Files
import kotlin.test.BeforeTest
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertNotNull
import kotlin.test.assertTrue

class ConfigurationTest : FileTestBase() {

    // SUT
    private lateinit var testConfiguration: ConfigurationProvider<TestConfiguration>

    @BeforeTest
    override fun setUp() {
        super.setUp()

        this.testConfiguration = ConfigurationProvider(this.tmpDir, TestConfiguration::class.java)
    }

    @Test
    fun hasConfigurationSaved() {
        // Given
        val configurationFile = this.tmpDir.resolve("testConfiguration.conf")

        // Then
        assertTrue(Files.exists(configurationFile))
    }

    @Test
    fun isConfigObjectLoadedProperly() {
        // When
        val config = this.testConfiguration.config

        // Then
        assertNotNull(config)
    }

    @Test
    fun isConfigVariablesLoadedProperly() {
        // Given
        val local = TestConfiguration()

        // When
        val config = this.testConfiguration.config

        // Then
        assertEquals(local.aString, config.aString)
        assertEquals(local.anInt.toLong(), config.anInt.toLong())
    }

    @Test
    fun variablesEditTest() {
        // Given
        val newInt = 6
        val newString = "Lorem ipsum dolor sit amet, consectetur."
        var config = this.testConfiguration.config

        // When
        config.anInt = newInt
        config.aString = newString
        this.testConfiguration.save()
        this.testConfiguration.reload()
        config = this.testConfiguration.config

        // Then
        assertEquals(newString, config.aString)
        assertEquals(newInt.toLong(), config.anInt.toLong())
    }
}
